kasa/aestransport.py

```diff
@@ -110,6 +121,19 @@ async def client_post(self, url, params=None, data=None, json=None, headers=None
 
         return resp.status_code, response_data
 
+    def _handle_response_error_code(self, resp_dict: dict, msg: str):
+        if (error_code := ErrorCode(resp_dict.get("error_code"))) != ErrorCode.SUCCESS:
+            msg = f"{msg}: {self.host}: {error_code.name}({error_code.value})"
+            if error_code in TIMEOUT_ERRORS:
+                raise TimeoutException(msg)
```

Comment:
I don't think we should have a separate custom exception for session timeouts, but rather try to reauthenticate to get a fresh session? The reason being is that I doubt any downstream users are interested on handling reauths manually, but are expecting that the library takes care of such low-level things.

Reply:
The protocol does do an automatic retry in `_query` when it catches these exceptions and also a small `asyncio.sleep()`.  I'm trying to separate the transport specific implementation details from the protocols.  To do more re-use of protocol code, in subsequent PR I want to have the transports only raise these custom exceptions so the protocols aren't catching transport specific exceptions like `httpx.TimeoutError`, `asyncio.TimeoutError` etc.  Then we can have a single implementation of `_query` with retries.  I wouldn't expect a downstream to have to handle these, only the protocols.

Reply:
Okay, we are not exposing them over the `kasa` package anyway, so I think it's good as it is. Thanks for the explanation!